### PR TITLE
Document remaining public APIs in embassy crate

### PR DESCRIPTION
**Pull request by lulf**
This also includes the README.md in the toplevel crate documentation, fixing a few formatting issues with it as well. 

Please review the mutex in detail in case I've misunderstood/missed a few things there.

### CODE REVIEW

File: embassy/src/blocking_mutex/mod.rs [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
 use core::cell::UnsafeCell;
 
 use self::raw::RawMutex;
 
-/// Any object implementing this trait guarantees exclusive access to the data contained
-/// within the mutex for the duration of the lock.
-/// Adapted from <https://github.com/rust-embedded/mutex-trait>.
+/// Blocking mutex (not async)
+///
+/// Provides a blocking mutual exclusion primitive backed by an implementation of [`raw::RawMutex`].
```
Comment (huntc) [inline embassy/src/blocking_mutex/mod.rs:12]: One thing that Dario mentioned, which helped me a lot, was that the blocking mutex is designed for bringing your own interior mutability. They are also low level on this basis. In particular, locks are intended to be short lived and must not cross async boundaries (and their design enforces this too). OTOH, the async mutex can have a lock spanning awaits, but also comes with a little more overhead.

**Review by huntc — APPROVED** [review]
Nice work! You beat me to it. :-)

File: embassy/src/blocking_mutex/mod.rs (hunk 1 above)
Comment (lulf) [reply embassy/src/blocking_mutex/mod.rs:12]: Thanks, I've emphasized this point

**Pull request merged**